### PR TITLE
Add credentials page to front-end AB#10578

### PR DESCRIPTION
**Pull request by MikeLyttle**
# Fixes or Implements [AB#10578](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10578)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Adds the credential page to the site.  It is accessible via the sidebar.  The page has two main states - viewing the instructions and managing the credentials - which are separated into different views.  There are two components utilized by the credential management view: the credential collection card, which displays information about the connection, and the credential list, which lists the various existing and eligible credentials.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.  Unit tests will be addressed in [AB#10658](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10658).

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts [hunk 1]
```diff
@@ -1,10 +1,7 @@
 import { injectable } from "inversify";
 
 import { ExternalConfiguration } from "@/models/configData";
-import {
-    CredentialConnection,
-    WalletCredential,
-} from "@/models/credentialConnection";
+import { WalletConnection, WalletCredential } from "@/models/credential";
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts:4]: can we change the name from credential to wallet?!

File: Apps/WebClient/src/ClientApp/src/views/credentials.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    @Action("getPatientData", { namespace: "user" })
+    getPatientData!: () => Promise<void>;
+
+    @Action("retrieve", { namespace: "immunization" })
+    retrieveImmunizations!: (params: { hdid: string }) => Promise<void>;
+
+    @Action("retrieveConnection", { namespace: "credential" })
+    retrieveConnection!: (params: { hdid: string }) => Promise<boolean>;
+
+    @Action("retrieveCredentials", { namespace: "credential" })
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/views/credentials.vue:50]: i think we might have to get rid of this, action and use the connection to retrieve the list of credentials

File: Apps/WebClient/src/ClientApp/src/components/credential/credentialCollectionCard.vue [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    }
+
+    private get connectionStatusVariant(): string | undefined {
+        if (this.connection?.state === ConnectionState.Connected) {
+            return "success";
+        }
+        return undefined;
+    }
+
+    private get connectionUndefined(): boolean {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/credential/credentialCollectionCard.vue:96]: nit isConnectionUndefined, same below

File: Apps/WebClient/src/ClientApp/src/components/credential/credentialList.vue [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+import { CredentialState, WalletCredential } from "@/models/credential";
+import { DateWrapper } from "@/models/dateWrapper";
+import { ImmunizationEvent } from "@/models/immunizationModel";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.container";
+import { ILogger } from "@/services/interfaces";
+
+library.add(faSyringe);
+
+interface Entry {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/credential/credentialList.vue:18]: nit: Entry sounds too general for me. Maybe CredentialEntry?

File: Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts:4]: IMO wallet is an implementation detail for the credential types

File: Apps/WebClient/src/ClientApp/src/views/credentials.vue (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/views/credentials.vue:50]: We cold use this if we want to refresh the credentials without having to retrieve the connection as well

**Review by FuriousLlama — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/credential/credentialList.vue (hunk 4 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/components/credential/credentialList.vue:18]: Sounds good, will update.

File: Apps/WebClient/src/ClientApp/src/components/credential/credentialCollectionCard.vue (hunk 3 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/components/credential/credentialCollectionCard.vue:96]: Will go with isConnectionUndefined.

File: Apps/WebClient/src/ClientApp/src/views/credentials.vue (hunk 2 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/views/credentials.vue:50]: Can discuss changing it later - right now it's out of scope of this task.

File: Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts (hunk 1 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/services/restCredentialService.ts]: Renamed the file to `wallet.ts` per discussion.

**Pull request merged**